src/storage/async_page_fetcher.hpp

```diff
@@ -27,18 +27,20 @@
 
 namespace cublog
 {
-
   class async_page_fetcher
   {
     public:
       using log_page_callback_type = std::function<void (const LOG_PAGE *, int)>;
       using data_page_callback_type = std::function<void (const FILEIO_PAGE *, int)>;
+      using log_boot_info_callback_type = std::function<void (std::string)>;
```

Comment:
Pass by reference.
```suggestion
      using log_boot_info_callback_type = std::function<void (const std::string&)>;
```

Reply:
Changed to std::string &&. Makes more sense as the value needs to cross thread stacks.